dipy/workflows/io.py

```diff
@@ -324,6 +332,259 @@ def run(self, input_files, vol_idx=0, out_dir="", out_split="split.nii.gz"):
             logging.info(f"Split volume saved as {osplit}")
 
 
+class ExtractB0Flow(Workflow):
+    @classmethod
+    def get_short_name(cls):
+        return "extract_b0"
+
+    def run(
+        self,
+        input_files,
+        bvalues_files,
+        bvectors_files,
+        b0_threshold=50,
+        bvecs_tol=0.01,
+        group_contiguous_b0=False,
+        strategy="mean",
+        chunk_size=None,
+        out_dir="",
+        out_b0="b0.nii.gz",
+    ):
+        """Extract on or multiple b0 volume from the input 4D file.
+
+        Parameters
+        ----------
+        input_files : string
+            Path to the input volumes. This path may contain wildcards to
+            process multiple inputs at once.
+        bvalues_files : string
+            Path to the bvalues files. This path may contain wildcards to use
+            multiple bvalues files at once.
+        bvectors_files : string
+            Path to the bvectors files. This path may contain wildcards to use
+            multiple bvectors files at once.
+        b0_threshold : float, optional
+            Threshold used to find b0 volumes.
+        bvecs_tol : float, optional
+            Threshold used to check that norm(bvec) = 1 +/- bvecs_tol
+        group_contiguous_b0 : bool, optional
+            If True, each contiguous b0 volumes are grouped together.
+        strategy : str, optional
+            The extraction strategy, of either:
+
+                - first: select the first b0 found.
+                - all: select them all.
+                - mean: average them.
+
+            When used in conjunction with the batch parameter set to True, the
+            strategy is applied individually on each continuous set found.
+        chunk_size : int, optional
+            Load the data in chunks of this size. Useful for handling large datasets
+            that cannot fit into memory at once.
+        out_dir : string, optional
+            Output directory. (default current directory)
+        out_b0 : string, optional
+            Name of the resulting b0 volume
+
+        """
+        io_it = self.get_io_iterator()
+        for dwi, bval, bvec, ob0 in io_it:
+            logging.info("Extracting b0 from {0}".format(dwi))
+            data, affine, image = load_nifti(dwi, return_img=True)
+
+            bvals, bvecs = read_bvals_bvecs(bval, bvec)
+            # If all b-values are smaller or equal to the b0 threshold, it is
+            # assumed that no thresholding is requested
+            if any(mask_non_weighted_bvals(bvals, b0_threshold)):
+                if b0_threshold < bvals.min():
```

Comment:
This warning is applicable to the original function as well. Maybe it should be in the base function and not the workflow?

Reply:
the original function expect b0_mask as an input which can be a simple boolean array.  No assumption and does not check the bvals. 

So I think, it is better here. I do not want to force the user to pass `bvals` in the original function.